### PR TITLE
Write DocumentAPI test data in two phases

### PR DESCRIPTION
**Pull request by vekterli**
@freva please review. Spuriously failing DocumentAPI tests during concurrent module builds indicate that cross-language tests can step on each other toes during file writing and reading. Now writing to temporary files and doing atomic renames, which should hopefully provide inode-level read isolation and stabilize things.

### CODE REVIEW

File: documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/TestFileUtil.java [hunk 1]
```diff
@@ -4,16 +4,27 @@
...
     protected static final String DATA_PATH = "./test/crosslanguagefiles";
 
     public static void writeToFile(String path, byte[] data) throws IOException {
-        try (FileOutputStream stream = new FileOutputStream(path)) {
+        // Write to a temporary file to avoid racing with cross-language tests reading the
+        // exact same file we're trying to write.
+        String tmpPath = path + ".tmp";
+        try (FileOutputStream stream = new FileOutputStream(tmpPath)) {
             stream.write(data);
+            stream.flush();
```
Comment (freva) [inline documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/TestFileUtil.java]: This doesn't do anything for `FileOutputStream` and for buffered writers `flush()` should be called by `close()` anyway.

**Review by freva — APPROVED** [review]
So I'm guessing only 1 of them _writes_ at the time and the other just did premature _reads_? Because otherwise the problem could persist if they write to the same filename + `.tmp` at the same time.

**Comment by vekterli** [comment]
The files written are distinct and their names are tagged with the originator language, so it's the first case you mention.

File: documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/TestFileUtil.java (hunk 1 above)
Comment (vekterli) [reply documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/TestFileUtil.java]: Removed

**Review by freva — APPROVED** [review]
:100:

**Comment by vekterli** [comment]
These changes were not sufficient, it is still possible to race and fail the rename operation itself. I'll revert this probably change the test logic to not even try writing the files unless they have actually changed (this kind of makes more sense in the first place...!)

**Pull request merged**